### PR TITLE
#9748 delete tools only added by tests

### LINKED ISSUE #9748: Installed External Tools deleted by running Integration Test
NOTE: This may not be worth fixing, but it did cause a minor annoyance while working on the Ask The Data tool PR. (I didn't put in the time then so as not to slow down the 5.14 release.) 

**What steps does it take to reproduce the issue?**
When you run ExternalToolsIT, the first thing it does is delete all of the existing external tools. You have to re-run the api to get your tools back.

* When does this issue occur?
Running the ExternalToolsIT

* Which page(s) does it occurs on?
it's manifested in the files list on the dataset page. Tool icons disappear after the test is run - and the external tools table is wiped out.

* What happens?

* To whom does it occur (all users, curators, superusers)?
developers

* What did you expect to happen?
it would be great if the tools didn't get wiped out, but that would probably make the existing tests fail - so fix the test so that the tools don't get wiped out.

**Which version of Dataverse are you using?**
dev/5.14

**Any related open or closed issues to this bug report?**
unknown

**Screenshots:**

No matter the issue, screenshots are always welcome.

To add a screenshot, please use one of the following formats and/or methods described here:

* https://help.github.com/en/articles/file-attachments-on-issues-and-pull-requests
*

### PR DESCRIPTION
**Pull request by sekmiller**
**What this PR does / why we need it**: Updates ExternalToolsIT so that existing tools are not deleted

**Which issue(s) this PR closes**:

Closes #9748 Installed External tools deleted by running ExternalToolsIT

**Special notes for your reviewer**: Going in I thought that deleting the tools was a requirement to get the tests to pass. That turned out not to be the case which made the fix easier. The tests were modified to delete only those tools added via the tests.

**Suggestions on how to test this**: Install an external tool in your test environment. Run ExternalToolsIT
verify that your pre-installed tool(s) exist after running the test. And that there are no additional tools.
api for getting a list of installed tools:
 curl http://localhost:8080/api/admin/externalTools

**Does this PR introduce a user interface change? If mockups are available, please link/include them here**: no

**Is there a release notes update needed for this change?**: no

**Additional documentation**: none

### CODE REVIEW

File: src/test/java/edu/harvard/iq/dataverse/api/ExternalToolsIT.java [hunk 1]
```diff
@@ -145,26 +130,14 @@ public void testFileLevelTool1() {
                 .statusCode(OK.getStatusCode())
                 // No tools for this file type.
                 .body("data", Matchers.hasSize(0));
+        
+        //Delete the tool added by this test...
+        Response deleteExternalTool = UtilIT.deleteExternalTool(toolId);
```
Comment (scolapasta) [inline src/test/java/edu/harvard/iq/dataverse/api/ExternalToolsIT.java:135]: @sekmiller for the other delete, you follow with an:
     deleteExternalTool.then().assertThat()
                .statusCode(OK.getStatusCode());
For consistency can you add that here?

**Review by scolapasta — CHANGES_REQUESTED** [review]

File: src/test/java/edu/harvard/iq/dataverse/api/ExternalToolsIT.java (hunk 1 above)
Comment (sekmiller) [reply src/test/java/edu/harvard/iq/dataverse/api/ExternalToolsIT.java:135]: sure

**Review by scolapasta — APPROVED** [review]

**Comment by pdurbin** [comment]
There's a test failure at https://jenkins.dataverse.org/job/IQSS-Dataverse-Develop-PR/job/PR-10274/1/testReport/edu.harvard.iq.dataverse.api/ExternalToolsIT/testFileLevelTool1/

Seems like an easy fix:

```
1 expectation failed.
JSON path data[0].displayName doesn't match.
Expected: AwesomeTool
  Actual: Data Explorer
```

**Comment by sekmiller** [comment]
Weird. It's not failing for me locally.

**Review by pdurbin — APPROVED** [review]
Tests are passing. Code looks good. Will merge.

**Pull request merged**